include/gfxbase.h

```diff
@@ -71,16 +71,22 @@
 #include "effects/matrix/Vector.h"
 #include <memory>
 
-typedef struct
-{
-    uint32_t noise_x;
-    uint32_t noise_y;
-    uint32_t noise_z;
-    uint32_t noise_scale_x;
-    uint32_t noise_scale_y;
-    uint8_t  noise[MATRIX_WIDTH][MATRIX_HEIGHT];
-    uint8_t  noisesmoothing;
-} Noise;
+#if USE_MATRIX
+#define USE_NOISE 1
+#endif
+
+#if USE_NOISE
```

Comment:
You only get the Noise functions for your GFXBase if you define this.  That way I can use this class for the screen drawing as well and not allocate huge noise tables on behalf of the large LCD.